src/mmrelay/matrix/sync_bootstrap.py

```
@@ -0,0 +1,1199 @@
+        passed_config (dict[str, Any] | None): Optional configuration override for this connection attempt; when provided it is used instead of the module-level configuration.
+
+    Returns:
+        AsyncClient | None: The configured and initialized AsyncClient on success; `None` if credentials, configuration, or connection setup failed and no client could be created.
+
+    Raises:
+        MissingMatrixRoomsError: If the required top-level "matrix_rooms" configuration is missing or invalid.
+        MatrixSyncTimeoutError: If the initial Matrix sync times out.
+        MatrixSyncFailedError: If the initial Matrix sync fails.
+        MatrixSyncFailedDetailsError: If the initial Matrix sync fails with detailed error information.
+    """
+    global matrix_client, bot_user_name, matrix_homeserver, matrix_rooms, matrix_access_token, bot_user_id
```

Ruff (0.15.9)

[warning] Using the global statement to update `matrix_client` is discouraged

(PLW0603)